app/src/main/java/com/github/orkest/bluetooth/ui/BluetoothActivity.kt

```diff
@@ -66,10 +84,18 @@ class BluetoothActivity() : ComponentActivity() {
 
                     return
                 } else {
-                    if (device != null) {
+                    if (device != null && device.name != "") {
```

Comment:
Another way to make sure that device.name is not null is by calling device.name.isNotNull()